frontend/src/components/pages/teacher/TeacherRouting.tsx

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { Redirect, Route, Switch } from "react-router-dom";
+import { Box, VStack } from "@chakra-ui/react";
+
+import * as Routes from "../../../constants/Routes";
+import Page from "../../../types/PageTypes";
+import PrivateRoute from "../../auth/PrivateRoute";
+import Navbar from "../../common/Navbar";
+import NotFound from "../NotFound";
+
+import ClassroomsPage from "./ClassroomsPage";
+import DistributeAssessmentPage from "./DistributeAssessmentPage";
+import TeacherDashboardPage from "./TeacherDashboardPage";
+
+const pages: Page[] = [
+  { title: "Dashboard", url: Routes.TEACHER_DASHBOARD_PAGE },
+  { title: "Assessments", url: Routes.DISTRIBUTE_ASSESSMENT_PAGE },
+  { title: "Classrooms", url: Routes.CLASSROOMS_PAGE },
+];
+
+const TeacherRouting = (): React.ReactElement => {
+  return (
+    <Switch>
+      <VStack align="left" flex="1">
+        <Navbar pages={pages} />
+        <Box padding="1.5em 2em 0em 2em">
+          <Switch>
+            <PrivateRoute
```

Comment:
unnecessary future refactor idea lol - we can create `TeacherRoute`, `AdminRoute`, and `StudentRoute` directly so we don't need `roles={["role"]}` everytime

Reply:
https://www.notion.so/uwblueprintexecs/Route-refactoring-352fae3f105e4c6bb0df065967ecd4d3?pvs=4 !